File: functions/func-fofoprono-synchro/src/models/database/game.ts
import { Team } from "./team";

export interface Game {
  id: number;
  time: Date;
  stage: Stage;

  team_home: Team;
  team_away: Team;

  odds_home?: number;
  odds_draw?: number;
  odds_away?: number;

  competition_id: number;
  external_api_id: number;
}

export type Stage = "group" | "sixteen" | "quarter" | "semi" | "final";

File: api/src/models/game.rs
use std::time::SystemTime;

use crate::schema::{competitions, games};
use diesel::prelude::*;
use diesel_derive_enum::DbEnum;
use serde::{Deserialize, Serialize};

#[derive(Identifiable, Queryable, Serialize, Deserialize)]
pub struct Game {
    pub id: i32,
    pub time: SystemTime,
    pub stage: Stage,

    pub team_home: String,
    pub team_away: String,

    pub score_home: Option<i32>,
    pub score_away: Option<i32>,

    pub odds_home: Option<f64>,
    pub odds_away: Option<f64>,
    pub odds_draw: Option<f64>,

    pub competition_id: i32,

    pub external_api_id: Option<i32>,
}

#[derive(Identifiable, Queryable, Serialize, Deserialize, Debug)]
pub struct Competition {
    pub id: i32,
    pub name: String,
}

#[derive(DbEnum, Serialize, Deserialize, Debug, Clone, Copy, PartialEq, Eq)]
#[ExistingTypePath = "crate::schema::sql_types::Stage"]
pub enum Stage {
    Group,
    Sixteen,
    Quarter,
    Semi,
    Final,
}
